packages/beacon-node/src/chain/opPools/opPool.ts

```diff
@@ -190,7 +220,17 @@ export class OpPool {
       }
     }
 
-    return [attesterSlashings, proposerSlashings, voluntaryExits];
+    const blsToExecutionChanges: capella.SignedBLSToExecutionChange[] = [];
+    for (const blsToExecutionChange of blsToExecutionChanges.values()) {
+      if (isValidBlsToExecutionChange(state as CachedBeaconStateCapella, blsToExecutionChange, false).valid) {
```

Comment:
Here we want to be extremely careful and potentially re-implement this function again thinking each condition. In the past we have bricked block production due to including bad objects. I would add a new function `isValidBlsToExecutionChangeForBlockProduction` or similar.

Lets think each condition from https://github.com/ethereum/consensus-specs/blob/dev/specs/capella/beacon-chain.md#new-process_bls_to_execution_change

1. `assert address_change.validator_index < len(state.validators)`: If valid before will always be valid in the future, no need to check
2. `assert validator.withdrawal_credentials[:1] == BLS_WITHDRAWAL_PREFIX`: Must be checked again, since it can already be changed by now.
3. `assert validator.withdrawal_credentials[1:] == hash(address_change.from_bls_pubkey)[1:]`: If valid before will always be valid in the future, no need to check
4. `assert bls.Verify(address_change.from_bls_pubkey, signing_root, signed_address_change.signature)`: May become invalid in the future if the fork changes. No need to check the full BLS signature, but check that the fork it was valid, is still the current fork

To prevent having to check this messages over and over they have to be pruned. Conditions 2. and 4. can be checked on finalized state and drop the no longer valid messages